src/main/java/org/jitsi/videobridge/eventadmin/callstats/ConferencePeriodicRunnable.java

```diff
@@ -0,0 +1,153 @@
+     */
+    private static final Logger logger
+        = Logger.getLogger(ConferencePeriodicRunnable.class);
+
+    /**
+     * The {@link MediaType}s for which we will report to callstats.
+     */
+    private static final MediaType[] MEDIA_TYPES
+        = { MediaType.AUDIO, MediaType.VIDEO };
+
+    /**
+     * Constructs <tt>ConferencePeriodicRunnable</tt>.
+     * @param conference the conference.
+     * @param period the reporting interval.
+     * @param statsService the StatsService to use for reporting.
+     * @param conferenceIDPrefix prefix to use when creating conference IDs.
+     * @param initiatorID the id which identifies the current bridge.
+     */
+    ConferencePeriodicRunnable(
+        Conference conference,
+        long period,
+        StatsService statsService,
+        String conferenceIDPrefix,
+        String initiatorID)
+    {
+        super(conference,
+            period,
+            statsService,
+            conference.getName(),
+            conferenceIDPrefix,
+            initiatorID);
+    }
+
+    @Override
+    protected Map<String, Collection<? extends ReceiveTrackStats>>
+        getReceiveTrackStats()
+    {
+        return getTrackStats(true);
+    }
+
+    @Override
+    protected Map<String, Collection<? extends SendTrackStats>>
+        getSendTrackStats()
+    {
+        return getTrackStats(false);
+    }
+
+    /**
+     * Get stats receive or send from current conference.
+     *
+     * @param receive whether to get receive if <tt>true</tt> or
+     * send statistics otherwise.
+     * @param <T> the type of result stats Collection, ReceiveTrackStats
+     * or SendTrackStats.
+     * @return the result collection of stats grouped by endpointID.
+     */
+    private <T extends Collection> Map<String, T> getTrackStats(boolean receive)
+    {
+        Map<String, T> resultStats = new HashMap<>();
+
+        for (Endpoint e : o.getEndpoints())
+        {
+            for (MediaType mediaType : MEDIA_TYPES)
+            {
+                for (RtpChannel channel : e.getChannels(mediaType))
+                {
+                    if (channel == null)
+                    {
+                        logger.debug("Could not log the channel expired event "
+                            + "because the channel is null.");
+                        continue;
+                    }
+
+                    if (channel.getReceiveSSRCs().length == 0)
+                    {
+                        continue;
+                    }
+
+                    MediaStream stream = channel.getStream();
+                    if (stream == null)
+                    {
+                        continue;
+                    }
+
+                    MediaStreamStats2 stats = stream.getMediaStreamStats();
+                    if (stats == null)
+                    {
+                        continue;
+                    }
+
+                    Endpoint endpoint = channel.getEndpoint();
```

Comment:
Why do we get the channel's endpoint instead of using "e"?

Reply:
Yep, I will change that. I was just moving the previous code here.

Reply:
Just forced pushed the changed.